### ISSUE TITLE
TypeScript claims “Cannot find modules ”

### ISSUE THREAD

**Issue by tats-u** [facebook/docusaurus#8683]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

In current Docusaurus version (2.x), no `declare module "*.png"`, `declare module "*.jpg"`, and so on are not defined and we get type erros when we try to import raster images unlike Next.js.
We have to avoid this kind of errors by adding these ambient declarations to `src/index.d.ts`.

To make matters worse, the `plugin-ideal-image` changes the type of imported supported images from `string` to `{src: {src: string, width: ...}, preSrc: string}`. We will have to select proper ambient declarations. Next.js resolves this problem by `next-env.d.ts` like:

```ts
/// <reference types="next" />
/// <reference types="next/image-types/global" />

// NOTE: This file should not be edited
// see https://nextjs.org/docs/basic-features/typescript for more information.
```

### Reproducible demo

https://codesandbox.io/s/docusaurus-bug-report1-kptbhq

### Steps to reproduce

1. Setup using TypeScript template
2. Add `import picture from "../../blog/2021-08-26-welcome/docusaurus-plushie-banner.jpeg";` and `<img src={picture} alt="" />` to `src/pages/index.tsx`

### Expected behavior

No type errors

### Actual behavior

```
Cannot find module '../../blog/2021-08-26-welcome/docusaurus-plushie-banner.jpeg' or its corresponding type declarations.ts(2307)
```

![image](https://user-images.githubusercontent.com/12870451/219943719-5ccee68d-90ee-4ef0-b677-fb241f855753.png)

### Your environment

- Public source code: https://codesandbox.io/s/docusaurus-bug-report1-kptbhq
- Docusaurus version used: 2.3.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4): CodeSandbox (Node 16) + TypeScript 4.9.5
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): CodeSandbox 

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Indeed, it seems we never expect someone to actually use import statements with `.png` files 😅 Our docs do recommend `src={require("...").default}` because it's several lines shorter.

I don't know if ambient declarations can override the default export. Do you know how this should work? Should we have `declare module "*.png"` under a separate named export and let users choose between the declarations from `module-name-aliases` or `plugin-ideal-image`?

**Comment by tats-u** [comment]
https://nextjs.org/docs/basic-features/typescript#existing-projects

- Next.js publishes image type declarations in `/image-types/global.d.ts`.
- Next.js can automatically generate `next-env.d.ts`. It says it should be ignored. However I have not understood how it's generated in Next.js CLI yet.

**Comment by slorber** [comment]
Note: the ideal plugin image returns `string | object` because in dev it can be disabled while it's always enabled in prod.

Due to the modular nature of Docusaurus, if we were to generate something similar to `next-env.d.ts` that would probably require a new plugin lifecycle so that the generated types are different if the image plugin is enabled/disabled.

I'd prefer to not rush into implementing this. Maybe let's see if we can find other plugins that would benefit from this.
Adding local typedefs yourself is a good temporary workaround.

**Comment by shadone** [comment]
Is importing raster images or generic files supposed to be working? I see that the documentation explicitly shows this to work https://docusaurus.io/docs/markdown-features/assets#images but it doesn't work for me, I get the same error as described in this issue

```ts
import myimage from "../../../static/img/hero.jpg";
```

> Cannot find module '../../../static/img/hero.jpg' or its corresponding type declarations.ts(2307)

**Comment by slorber** [comment]
Importing jpg/png works, it's just TypeScript that complains. 

We are still unsure how to fix the TS errors. In the meantime you can just add `@ts-expect-error` where TS complains
